clients/src/main/java/org/apache/kafka/common/requests/BrokerRegistrationRequest.java

```diff
@@ -45,7 +46,21 @@ public short oldestAllowedVersion() {
 
         @Override
         public BrokerRegistrationRequest build(short version) {
-            return new BrokerRegistrationRequest(data, version);
+            if (version < 4) {
+                // Workaround for KAFKA-17011: for BrokerRegistrationRequest versions older than 4,
+                // exclude support version ranges that begin with 0.
```

Comment:
I think we still need to finalize/agree here about omission vs. using 1 for older versions of the api. 

Right now we have omit for broker registration, but using version 1 for ApiVersions. We also have the option to use 1 for both or omit for both. Note: In 3.8, we omit for both requests.

I think there is some benefit in not including a version with 0 to older controllers rather than 1, but can also understand that it is confusing to have different approaches for different APIs. 

@junrao I know you had some opinions here.

Reply:
It seems that it's simpler for the old BrokerRegistrationRequest to be consistent with the older version of ApiVersionResponse, i.e., to include the feature but set the minSupportedVersion to 1, instead of 0.

Reply:
OK. Just like in `ApiVersionsResponse`, we will translate min version = 0 to min version = 1 here.